jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java

```diff
@@ -195,10 +195,10 @@ static Multimap<String, String> decodeTokenRepositoryGrants(String token) {
     // parts (header, payload, signature), collated with a ".".  The header and payload are
     // JSON objects.
     String[] jwtParts = token.split("\\.", -1);
-    byte[] payloadData;
-    if (jwtParts.length != 3 || (payloadData = Base64.decodeBase64(jwtParts[1])) == null) {
+    if (jwtParts.length != 3) {
```

Comment:
The original code returns null if input is null. Since jwtParts[1] cannot be null ([String.split](https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#split(java.lang.String,%20int)) doesn't return null strings) the output cannot be null.
Additionally `Base64.getDecoder().decode(` is null-hostile, so it would throw on null inputs and never return null.